client/coral-embed/index.js

```diff
@@ -0,0 +1,163 @@
+/* eslint-disable no-var */
+/**
+ * @file Exposes the main 'window.Coral.Talk' API that developers can use to
+ * render Talk streams in their webpages
+ * @todo Currently implemented to be included directly via <script> tag
+ * (including copypasta dependencies like pym.js), but later there will be a
+ * build step and this code may use import statements 
+ */
+// using umd.js (https://github.com/umdjs/umd/blob/master/templates/returnExports.js)
+(function (root, factory) {
+  /* eslint-disable */
+  if (typeof define === 'function' && define.amd) {
+
+    // AMD. Register as an anonymous module.
+    define([], factory);
+  } else if (typeof module === 'object' && module.exports) {
+
+    // Node. Does not work with strict CommonJS, but
+    // only CommonJS-like environments that support module.exports,
+    // like Node.
+    module.exports = factory();
+  } else {
+
+    // Browser globals (root is window)
+    root.Coral = factory();
+  }
+  /* eslint-enable */
+}(this, function () {
+
+  // This function should return value of window.Coral
+  let pym = requirePym();
+  let Coral = {};
+  let Talk = Coral.Talk = {};
+
+  /**
+   * Render a Talk stream
+   * @param {HTMLElement} el - Element to render the stream in
+   * @param {Object} opts - Configuration options for talk
+   * @param {String} opts.talk - Talk base URL
+   * @param {String} [opts.title] - Title of Stream (rendered in iframe)
+   * @param {String} [opts.asset] - parent Asset ID or URL. Comments in the
+   * stream will replies to this asset
+   */
+  Talk.render = function (el, opts) {
+    if ( ! el) {
+      throw new Error('Please provide Coral.Talk.render() the HTMLElement you want to render Talk in.');
+    }
+    if (typeof el !== 'object') {
+      throw new Error(`Coral.Talk.render() expected HTMLElement but got ${  el  } (${  typeof el})`);
+    }
+    opts = opts || {};
+
+    // @todo infer this URL without explicit user input (if possible, may have to be added at build/render time of this script)
+    if (! opts.talk) {
+      throw new Error('Coral.Talk.render() expects opts.talk as the Talk Base URL');
+    }
+
+    // ensure el has an id, as pym can't directly accept the HTMLElement
+    if ( ! el.id) {el.id = `_${  String(Math.random())}`;}
+    let asset = opts.asset || window.location;
```

Comment:
We have to support "asset_id + asset_url" or just "asset_url". This might require further expansion here.